### ISSUE TITLE
Melhorar interface mapserver

### ISSUE THREAD

**Issue by claudia-codeco** [AlertaDengue/AlertaDengue#183]
# Objetivos / Objectivos / Purpose

Melhorar a usabilidade do formulário e consertar camadas inoperantes.

## Geral / General

* Objetivo da solicitação / Objectivo de la solicutud / Purpose of the request

## Específicos / Specific

- Na caixa de busca, permitir busca simultâneamente por geocódigo, Nome de município e nome de estado.
mostrando as três informações juntas na seleção. Por exemplo: "3304557 - Rio de janeiro - RJ"
- Permitir a seleção de multiplos polígonos de uma só vez.
- Criar um preview da seleção ao lado do formulário em um mapa leaflet.
- Gerar uma página de erro informativa, quando a camada não puder ser servida.
- Adicionar caixa de data para especificar a que semana se referem as camadas servidas.

**Comment by xmnlab** [comment]
@claudia-codeco 

> Permitir a seleção de multiplos polígonos de uma só vez.

Essa opção seria a opção de mapas? (LST Dia, LST noite, Dengue, etc)

Se sim, cada mapa é uma URL diferente, então não funcionaria.

Se essa seleção é em relação a cidade, OK.

> Adicionar caixa de data para especificar a que semana se referem as camadas servidas.

Atualmente está sendo gerada apenas para a última semana de dados. Isso foi feito por falta de especificação (https://github.com/AlertaDengue/AlertaDengue/issues/115#issuecomment-358653141).

Basicamente, para cada semana, vamos precisar criar um mapa novo, creio que seria importante criar uma issue para pedir essa funcionalidade e lá discutimos para definir a estratégia que iremos tomar para resolver o tema.

**Comment by claudia-codeco** [comment]
@xmnlab 

Sim, isso, permitir a seleção de multiplas cidades para gerar por exemplo, mapas de temperatura para uma região inteira. Bastaria permitir escrever uma lista de geocodigos. 

Entendi, sobre a data, farei a issue. Mesmo assim, é importante identificar em algum lugar (a seu critério) que o mapa gerado corresponde a semana epidemiologica SSYYYY, ok?

**Comment by fccoelho** [comment]
A seleção múltipla é relacionada às cidades.

Com relação a oferecer outras semanas de mapas, podemos abrir nova issue
sim. Você pode abrir?

2018-04-17 23:16 GMT-03:00 Ivan <notifications@github.com>:

**Comment by xmnlab** [comment]
> Entendi, sobre a data, farei a issue. Mesmo assim, é importante identificar em algum lugar (a seu critério) que o mapa gerado corresponde a semana epidemiologica SSYYYY, ok?

adicionando isso no mapserver, a imagem retornada será relacionado a SSAAAA especificado, caso não houver, vai retornar um erro:

Erro se não houver layer:

https://maps.dengue.mat.br/?map=/maps/alert/dengue.map&SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&LAYERS=999993304557&STYLES=&CRS=CRS:84&BBOX=-43.79653853090349,-23.0828926973403,-43.09903940038369,-22.746020285810395&WIDTH=800&HEIGHT=400&FORMAT=image/png

erro se não houver o mapa:

https://maps.dengue.mat.br/?map=/maps/alert/dengue9999.map&SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&LAYERS=3304557&STYLES=&CRS=CRS:84&BBOX=-43.79653853090349,-23.0828926973403,-43.09903940038369,-22.746020285810395&WIDTH=800&HEIGHT=400&FORMAT=image/png

**Comment by luabida** [comment]
Closing as not followed. Can be reopened if necessary
